### PR TITLE
[#171211343] - user data processing api

### PR DESCRIPTION
**Pull request by AleDore**
Refactoring UserDataProcessing API fields exposure

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#171211343](https://www.pivotaltracker.com/story/show/171211343): Raccogliere la volontà dell'utente di veder cancellato il proprio account e avere a disposizione l'esportazione dei propri dati

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by gunzip** [comment]
it looks like this needs rebasing (moreover tests does not pass)

File: azure-functions-core-tools/.telemetry/20200228172129_dead316098664637afd7e6a17ef62ed3.trn [hunk 1]
```diff
@@ -0,0 +1,5 @@
+https://dc.services.visualstudio.com/v2/track
```
Comment (gunzip) [inline azure-functions-core-tools/.telemetry/20200228172129_dead316098664637afd7e6a17ef62ed3.trn]: can you please remove this file and put the whole dir into gitignore ?

File: utils/user_data_processings.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { UserDataProcessing } from "io-functions-commons/dist/generated/definitions/UserDataProcessing";
+import { RetrievedUserDataProcessing } from "io-functions-commons/dist/src/models/user_data_processing";
+
+/**
+ * Converts a RetrievedProfile model to an ExtendedProfile
+ */
+export function retrievedUserDataProcessingToUserDataProcessingApi(
```
Comment (gunzip) [inline utils/user_data_processings.ts]:
```suggestion
export function toUserDataProcessingApi(
```

suffices

**Comment by BurnedMarshal** [comment]
# [Codecov](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=h1) Report
> Merging [#40](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=desc) into [master](https://codecov.io/gh/pagopa/io-functions-app/commit/5e336be3833858a10e4ca82a94b960d7fb293199&el=desc) will **increase** coverage by `0.17%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #40      +/-   ##
==========================================
+ Coverage   80.86%   81.04%   +0.17%     
==========================================
  Files          18       19       +1     
  Lines         533      538       +5     
  Branches       51       51              
==========================================
+ Hits          431      436       +5     
  Misses        102      102              
```


| [Impacted Files](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [utils/user\_data\_processings.ts](https://codecov.io/gh/pagopa/io-functions-app/pull/40/diff?src=pr&el=tree#diff-dXRpbHMvdXNlcl9kYXRhX3Byb2Nlc3NpbmdzLnRz) | `100.00% <0.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=footer). Last update [5e336be...a032de3](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: utils/user_data_processings.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { UserDataProcessing } from "io-functions-commons/dist/generated/definitions/UserDataProcessing";
+import { RetrievedUserDataProcessing } from "io-functions-commons/dist/src/models/user_data_processing";
+
+/**
+ * Converts a RetrievedProfile model to an ExtendedProfile
```
Comment (gunzip) [inline utils/user_data_processings.ts]:
```suggestion
 * Converts a UserDataProcessingProfile model to an UserDataProcessing
```

**Comment by BurnedMarshal** [comment]
# [Codecov](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=h1) Report
> Merging [#40](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=desc) into [master](https://codecov.io/gh/pagopa/io-functions-app/commit/5e336be3833858a10e4ca82a94b960d7fb293199&el=desc) will **increase** coverage by `0.17%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #40      +/-   ##
==========================================
+ Coverage   80.86%   81.04%   +0.17%     
==========================================
  Files          18       19       +1     
  Lines         533      538       +5     
  Branches       51       51              
==========================================
+ Hits          431      436       +5     
  Misses        102      102              
```


| [Impacted Files](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [utils/user\_data\_processings.ts](https://codecov.io/gh/pagopa/io-functions-app/pull/40/diff?src=pr&el=tree#diff-dXRpbHMvdXNlcl9kYXRhX3Byb2Nlc3NpbmdzLnRz) | `100.00% <0.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=footer). Last update [5e336be...eb29d13](https://codecov.io/gh/pagopa/io-functions-app/pull/40?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**